vnext/Microsoft.ReactNative/Views/VirtualTextViewManager.h

```diff
@@ -16,17 +17,15 @@ struct VirtualTextShadowNode final : public ShadowNodeBase {
   TextTransform textTransform{TextTransform::Undefined};
 
   void AddView(ShadowNode &child, int64_t index) override;
+  void RemoveChildAt(int64_t indexToRemove) override;
+  void removeAllChildren() override;
 
-  static void ApplyTextTransform(ShadowNodeBase &node, TextTransform transform, bool forceUpdate, bool isRoot);
+  void NotifyAncestorsTextPropertyChanged(TextViewManager::PropertyChangeType propertyChangeType);
 
-  struct HighlightData {
-    std::vector<HighlightData> data;
```

Comment:
I'm missing where we point to the highlights for the children; is it no longer needed because now we do `NotifyAncestorsTextPropertyChanged`?

Reply:
Yeah. Each `VirtualTextShadowNode` maintains its foreground color and background color. If either of these change, we call `NotifyAncestorsTextPropertyChanged`, which traverses to the root text to trigger highlighter recalculation.

Reply:
Which makes me wonder if we want to create an optimization for `backgroundColor` and `color` prop changes. We could store the `TextHightlighter` in the `VirtualTextShadowNode`, and if only the prop value changes (the children don't change), we just update the brushes directly on the `TextHighlighter`.

I guess it's just a question of whether we think that's a common enough scenario to optimize for. The current design is "universal" for the relevant mutations of the text (text nodes added, text nodes removed, background color changed, foreground color changed, etc.), but not necessarily optimal.

Reply:
I think the optimization would add too much complexity. It's already complex enough with the optimization to only traverse the text tree when needed.